packages/main/src/components/protected/sidebar/components/DropRoom.jsx

```diff
@@ -1,10 +1,59 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "../styles/Drop.module.css";
 import ToggleSwitch from "./ToggleSwitch";
 import { BsToggleOff } from "react-icons/bs";
 import GrCircleAlert from "react-icons/gr";
+import axios from "axios";
 
 const DropRoom = props => {
+  //{ updatePrivacy, createChannel } = useChannel(props.data?.room_id);
+  const [state, setState] = useState({});
+
+  const handleEvent = ({ target }) => {
+    setState({
+      ...state,
+      [target.name]: target.type === "checkbox" ? target.checked : target.value
+    });
+  };
+
+  const handleSubmission = e => {
+    e.preventDefault();
+    //createChannel(state);
+    console.log(state);
+  };
+  const org_id = localStorage.getItem("currentWorkspace");
+  const user = JSON.parse(sessionStorage.getItem("organisations"));
+  const room = window.location.href.split("/").at(6);
+  const BASE_URL = "https://chat.zuri.chat";
+  const tina = user?.find(x => x.id == org_id)?.member_id;
+  const [data, setData] = useState(null);
+
+  useEffect(() => {
+    axios
+      .get(`${BASE_URL}/api/v1/org/${org_id}/rooms/${room}`)
+      .then(res => {
+        setData(res.data.data);
+      })
+      .catch(e => console.log(e));
+  }, [5]);
+
+  const updatePrivacy = () => {
+    const tina = user?.find(x => x.id === org_id)?.member_id;
+    if (data !== null) {
+      const newData = { ...data, is_private: JSON.stringify(state.Accept) };
+      console.log(`this is a new ${newData}`);
+      axios
+        .put(
+          `${BASE_URL}/api/v1/org/${org_id}/members/${tina}/rooms/${room}`,
+          newData
+        )
+        .then(res => {
+          console.log(res);
+        })
+        .catch(e => console.log(e));
```

Comment:
handle the error appropriately use a console error and then display a toast notification to let the user know